Modified MuJoCo env to be compatible with MuJoCo200

Rowing0914 opened the pull request:
Hi guys,

Thank you for this great repo!
Since I had to work on some experiments myself, I modified the repo to the one which is compatible with the latest MuJoCo version 2.0(https://github.com/openai/mujoco-py), which has been released since March in 2019.

So, I hope this enhances more users.

Rowing0914 committed:
modified MuJoCo env to be compatible with MuJoCo200

Breakend commented:
This is great! Thanks of the update. I think to make sure these changes don't affect reproducibility, we should bump all the version numbers of affected MuJoCo envs to -V1 in the registration file as well maybe?

Rowing0914 commented:
Thank you for your prompt response!
that's nice!! I'll do that and re-push it!

Rowing0914 committed:
change the naming rule of env name: v0(MuJoCo<2.0ver) to v1(MuJoCo==ver2.0)

Rowing0914 commented:
To test all mujoco envs, please run `tests/all_tests.py`(https://github.com/Rowing0914/gym-extensions/blob/mujoco200/tests/all_tests.py)

Breakend commented:
Thanks! Can you remove the .idea folder that got added. After that I'll run some tests to sanity check that the envs seem ok and an RL agent can still learn on them. Then I'll merge the pull request.

Rowing0914 committed:
deleted .idea dir

Rowing0914 commented:
Oh, snap.. deleted it!
Thank you for letting me know about that!

Rowing0914 committed:
Added the food env in where the ant agnet seeks the food

Rowing0914 committed:
added .idea in gitignore

Rowing0914 commented:
I have added the food env, which is a similar concept to AnimalAI(http://animalaiolympics.com/)!! I hope you like it.

Rowing0914 committed:
forgot to remove .idea, but it's already in .gitignore so that this will never happen again

Rowing0914 committed:
Added the envs created for NerveNet experiments

Rowing0914 commented:
I have added extra envs which have been mainly created for NerveNet,https://github.com/WilsonWangTHU/NerveNet, experiments purpose. I modified them to be compatible with MuJoCo ver 2.0, and named them by `v1` as we agreed! I hope you like them!

Rowing0914 committed:
modified README and moved the test files to ./tests

Rowing0914 commented:
This is the list for newly added envs.

```python
"""
ALL_ENV_LIST = [
    WalkersHopperone-v1
    WalkersHalfhumanoidone-v1
    WalkersHalfcheetahone-v1
    WalkersFullcheetahone-v1
    WalkersOstrichone-v1
    WalkersHoppertwo-v1
    WalkersHalfhumanoidtwo-v1
    WalkersHalfcheetahtwo-v1
    WalkersFullcheetahtwo-v1
    WalkersOstrichtwo-v1
    WalkersHopperthree-v1
    WalkersHalfhumanoidthree-v1
    WalkersHalfcheetahthree-v1
    WalkersFullcheetahthree-v1
    WalkersOstrichthree-v1
    WalkersHopperfour-v1
    WalkersHalfhumanoidfour-v1
    WalkersHalfcheetahfour-v1
    WalkersFullcheetahfour-v1
    WalkersOstrichfour-v1
    WalkersHopperfive-v1
    WalkersHalfhumanoidfive-v1
    WalkersHalfcheetahfive-v1
    WalkersFullcheetahfive-v1
    WalkersOstrichfive-v1
    CentipedeThree-v1
    CentipedeFour-v1
    CentipedeFive-v1
    CentipedeSix-v1
    CentipedeSeven-v1
    CentipedeEight-v1
    CentipedeTen-v1
    CentipedeTwelve-v1
    CentipedeFourteen-v1
    CentipedeTwenty-v1
    CentipedeThirty-v1
    CentipedeForty-v1
    CentipedeFifty-v1
    ReacherZero-v1
    ReacherOne-v1
    ReacherTwo-v1
    ReacherThree-v1
    ReacherFour-v1
    ReacherFive-v1
    ReacherSix-v1
    ReacherSeven-v1
    SnakeThree-v1
    SnakeFour-v1
    SnakeFive-v1
    SnakeSix-v1
    SnakeSeven-v1
    SnakeEight-v1
    SnakeNine-v1
    SnakeTen-v1
    SnakeTwenty-v1
    SnakeForty-v1
]
"""
```

and you can test them all from `./tests/nervenet_env_all_test.py`!!

zengrui324 commented:
My mujoco environment is 2.0 and python=3.6.2, but it prompts me "You appear to be missing Mujoxo. We expected to find the file here:/home/**/.mujoco/mjpro150"

Rowing0914 commented:
@zengrui324 sorry, I haven't worked on this project since my last commit...
Hope someone can help you...
From my memory, maybe you can double check the mujoco path things...
